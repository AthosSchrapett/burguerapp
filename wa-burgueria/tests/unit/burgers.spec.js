import mockAxios from "jest-mock-axios";
import BurgersService from "@/services/burgers";

describe("IngredientsService 🍔🍔🍔", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should return a list of burgers from backend", async () => {
    const burgers = [
      {
        nome: "Mariane Squenquel Peixe",
        carne: "Alcatra",
        pao: "Integral",
        opcionais: ["Bacon", "Salame"],
        status: "Solicitado",
        id: 1,
      },
      {
        nome: "Sharpzin",
        carne: "Picanha",
        pao: "3 Queijos",
        opcionais: ["Cheddar", "Tomate", "Pepino"],
        status: "Finalizado",
        id: 2,
      },
    ];
    mockAxios.get.mockImplementationOnce(() => {
      return Promise.resolve({ data: { burgers } });
    });

    const response = await BurgersService(mockAxios).getBurgers();

    expect(response.data).toHaveProperty("burgers");
    expect(response.data).toMatchSnapshot();
  });

  it("should create a Order with a burger product", async () => {
    const burger = {
      nome: "Jose Cabaninha",
      carne: "Picanha",
      pao: "Integral",
      opcionais: ["Cheddar", "Tomate"],
      status: "Solicitado",
      id: 3,
    };

    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: burger });
    });

    const response = await BurgersService(mockAxios).createOrder({ burger });

    expect(response.data).toHaveProperty(
      "nome",
      "pao",
      "carne",
      "opcionais",
      "status"
    );
    expect(response.data).toMatchSnapshot();
  });
});
