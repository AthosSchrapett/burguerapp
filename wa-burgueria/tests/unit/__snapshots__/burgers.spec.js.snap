// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientsService 🍔🍔🍔 should create a Order with a burger product 1`] = `
Object {
  "carne": "Picanha",
  "id": 3,
  "nome": "Jose Cabaninha",
  "opcionais": Array [
    "Cheddar",
    "Tomate",
  ],
  "pao": "Integral",
  "status": "Solicitado",
}
`;

exports[`IngredientsService 🍔🍔🍔 should return a list of burgers from backend 1`] = `
Object {
  "burgers": Array [
    Object {
      "carne": "Alcatra",
      "id": 1,
      "nome": "Mariane Squenquel Peixe",
      "opcionais": Array [
        "Bacon",
        "Salame",
      ],
      "pao": "Integral",
      "status": "Solicitado",
    },
    Object {
      "carne": "Picanha",
      "id": 2,
      "nome": "Sharpzin",
      "opcionais": Array [
        "Cheddar",
        "Tomate",
        "Pepino",
      ],
      "pao": "3 Queijos",
      "status": "Finalizado",
    },
  ],
}
`;
